index.html

```diff
@@ -1028,6 +1061,461 @@ <h2>
+        slots described in the following table:
+      </p>
+      <table class="simple">
+        <tr>
+          <th>
+            Internal slot
+          </th>
+          <th>
+            Initial value
+          </th>
+          <th>
+            Description
+          </th>
+        </tr>
+        <tr>
+          <td>
+            <dfn data-dfn-for="GamepadHapticActuator">[[\effects]]</dfn>
+          </td>
+          <td>
+            An empty [=list=] of {{Gamepad/GamepadHapticEffectType}}.
+          </td>
+          <td>
+            Represents the effects supported by the actuator.
+          </td>
+        </tr>
+        <tr>
+          <td>
+            <dfn data-dfn-for=
+            "GamepadHapticActuator">[[\playingEffectPromise]]</dfn>
+          </td>
+          <td>
+            `null`
+          </td>
+          <td>
+            The {{Promise}} for the playing effect, or `null` if no effect is
+            playing.
+          </td>
+        </tr>
+      </table>
+      <dl data-dfn-for="GamepadHapticActuator">
+        <dt>
+          <dfn>effects</dfn> attribute
+        </dt>
+        <dd>
+          <p>
+            Array of {{Gamepad/GamepadHapticEffectType}} values representing all
+            the types of haptic effects that the actuator supports. This
+            property MUST contain all the {{Gamepad/GamepadHapticEffectType}}
+            values that the actuator supports, unless the [=user agent=] does
+            not support playing effects of that type. The same object MUST be
+            returned and MUST NOT change for the lifetime of the
+            {{GamepadHapticActuator}}.
+          </p>
+          <p>
+            The {{GamepadHapticActuator/effects}} getter steps are:
+          </p>
+          <ol>
+            <li>Return [=this=].{{GamepadHapticActuator/[[effects]]}}.
+            </li>
+          </ol>
+        </dd>
+        <dt>
+          <dfn>playEffect()</dfn> method
+        </dt>
+        <dd>
+          <p>
+            The {{GamepadHapticActuator/playEffect()}} method steps, called with
+            {{GamepadHapticEffectType}} |type:GamepadHapticEffectType| and
+            {{GamepadEffectParameters}} |params:GamepadEffectParameters |, are:
+          </p>
+          <ol class="algorithm">
+            <li>If |params:GamepadEffectParameters| does not describe a [=valid
+            effect=] of type |type:GamepadHapticEffectType|, return [=a promise
+            rejected with=] reason {{TypeError}}.
+            </li>
+            <li>Let |document:Document?| be the [=current settings object=]'s 
+            [=relevant global object=]'s [=associated `Document`=].
+            </li>
+            <li>If |document| is `null` or |document| is not [=Document/fully
+            active=] or |document|'s [=Document/visibility state=] is
+            `"hidden"`, return [=a promise resolved with=]
+            {{GamepadHapticsResult/"preempted"}}.
```

Comment:
These still look like exceptional situations to me... we should be retiring [=a promise rejected with=] with something here.

Reply:
Will edit to return a promise rejected with an "{{InvalidStateError}}" {{DOMException}}.